examples/reporting/parallel_report_download.rb

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/ruby
+# Encoding: utf-8
+#
+# Copyright:: Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Shows how to download a set of reports from a list of accounts in parallel.
+# If you need to obtain a list of accounts, please see get_account_hierarchy.rb
+# or list_accessible_customers.rb examples.";
+
+require 'optparse'
+require 'google/ads/google_ads'
+require 'thread'
+
+def parallel_report_download(customer_ids, login_customer_id)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  # Optional login customer ID if your access to the CIDs is
+  # via a manager account.
+  client.configure do |config|
+    if login_customer_id
+      config.login_customer_id = login_customer_id.tr("-", "").to_i
+    end
+  end
+
+  ga_service = client.service.google_ads
+
+  query_list = {}
```

Comment:
Based on the way you're using query_list below, this should actually be an array that stores 2 pieces of information per element, rather than a map of key->value. Can you refactor it to be an array instead?

Reply:
Changed to an array.